File: Frontend/src/app/services/tax.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Tax } from '../models/tax-management.model';

@Injectable({
  providedIn: 'root'
})
export class TaxService {
  private apiUrl = 'http://127.0.0.1:8081/taxmanagement/api/taxes';

  constructor(private http: HttpClient) { }

  getAllTaxes(): Observable<Tax[]>{
    return this.http.get<Tax[]>(this.apiUrl);
  }

  getTaxById(id: number): Observable<Tax> {
   return this.http.get<Tax>(`${this.apiUrl}/${id}`);
  }

  createTax(tax: Tax): Observable<Tax> {
    return this.http.post<Tax>(this.apiUrl,tax);
  }

  updateTax(id: number, tax: Tax): Observable<Tax> {
    return this.http.put<Tax>(`${this.apiUrl}/${id}`, tax);
  }

  deleteTax(id: number):Observable<any> {
    return this.http.delete(`${this.apiUrl}/${id}`);
  }
}

File: Backend/src/main/java/com/taxmanagement/service/TaxService.java
package com.taxmanagement.service;

import java.util.List;

import com.taxmanagement.dto.TaxDTO;

public interface TaxService {
	TaxDTO createTax(TaxDTO taxDTO);

	boolean deleteTaxById(Long id);

	List<TaxDTO> getAllTaxes();

	TaxDTO getTaxById(Long id);

	TaxDTO updateTax(TaxDTO taxDTO);
}
